### PR TITLE
FIX upgrade mosquitto lib from 1.5.0 to 2.0.11

### PR DESCRIPTION
**Pull request by fgalan**
Part of PR #3888

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -226,6 +226,7 @@ find_package (mongoc-1.0 1.17.4 EXACT)
 # Static libs common to contextBroker and unitTest binaries
 SET (COMMON_STATIC_LIBS
     microhttpd.a
+    mosquitto.a
```
Comment (fgalan) [inline CMakeLists.txt:229]: Not using static linking makes RPM stuff harder (as the RPM would need libmosquitto as dependency and it seems it is not available at yum)

```
[fermin@centocho fiware-orion]$ cat /etc/redhat-release 
CentOS Linux release 8.3.2011
[fermin@centocho fiware-orion]$ sudo yum search mosquitto
Última comprobación de caducidad de metadatos hecha hace 0:12:39, el lun 09 ago 2021 07:25:55 EDT.
No se encontraron coincidencias.
```

Comment (fgalan) [reply CMakeLists.txt:229]: From https://pkgs.org/search/?q=mosquitto

imagen

So maybe we could get and old version of mosquitto (via EPEL), but better to use the newest one (2.0.11)

Comment (fgalan) [reply CMakeLists.txt:229]: NTC (informative)

**Comment by fgalan** [comment]
RPM installation worded (from commit 7d9fe24):

```
[fermin@centocho fiware-orion]$ make rpm
...
[fermin@centocho fiware-orion]$ cd test/check_rpm
[fermin@centocho check_rpm]$ WORKSPACE=/home/fermin/src/fiware-orion/ ./check_rpm.sh
[07:25:21] [INFO] ######## Executing the Check RPM Stage ... ########

[07:25:21] [INFO] ######## Checking MongoDB... ########
[07:25:21] #### Check the MongoDB process... ####
[07:25:21] #### Checking the mongodb status... #####
Redirecting to /bin/systemctl status mongod.service
[07:25:24] [OK] .............. Mongod is running ..............

[07:25:24] [INFO] ######## Installing RPM contextBroker ... ########
Verifying...                          ################################# [100%]
Preparando...                         ################################# [100%]
Actualizando / instalando...
   1:contextBroker-tests-3.1.0_next-de################################# [ 50%]
useradd: aviso: el directorio personal ya existe.
No se va a copiar ningún fichero del directorio «skel» en él.
   2:contextBroker-3.1.0_next-dev     ################################# [100%]
contextBroker requires additional configuration before the service can be
started. Edit '/etc/sysconfig/contextBroker' to provide the needed database
configuration.

Note that if you have a previously existing '/etc/sysconfig/contextBroker' it
has been renamed to /etc/sysconfig/contextBroker.orig-2021-08-09.

After configuring /etc/sysconfig/contextBroker execute 'chkconfig contextBroker on' to
enable contextBroker after a reboot.
[07:25:31] [OK] .............. contextBroker RPM installed ..............

[07:25:31] [INFO] ######## Checking the user configuration... ########
[07:25:31] [OK] .............. User orion is created ..............

[07:25:31] [INFO] ######## Check contextBroker directories... ########
[07:25:31] [OK] .............. directories are created well ..............

[07:25:31] [INFO] ######## Starting the contextBroker service... ########
contextBroker is stopped
Starting...
Starting contextBroker...                                  [  OK  ]
[07:25:34] [OK] .............. contextBroker service is started ..............

[07:25:34] [INFO] ######## Checking the contextBroker service ########
[07:25:34] [OK] .............. contextBroker service is running ..............

[07:25:34] [INFO] ######## Checking the contextBroker service ########
[07:25:34] [OK] .............. contextBroker is LISTEN the correct ports! ..............

[07:25:34] [INFO] ######## Starting the END-TO-END Check ########
HTTP/1.1 200 OK
Connection: Keep-Alive
Content-Length: 699
Content-Type: application/json
Fiware-Correlator: 83a02dfa-f904-11eb-8fb9-000c29d07fac
Date: Mon, 09 Aug 2021 11:25:34 GMT

{
"orion" : {
  "version" : "3.1.0-next",
  "uptime" : "0 d, 0 h, 0 m, 2 s",
  "git_hash" : "7d9fe2423c4c91efb587cadacdd38fcd7aebfb1a",
  "compile_time" : "Mon Aug 9 07:24:08 EDT 2021",
  "compiled_by" : "fermin",
  "compiled_in" : "centocho",
  "release_date" : "Mon Aug 9 07:24:08 EDT 2021",
  "machine" : "x86_64",
  "doc" : "https://fiware-orion.rtfd.io/",
  "libversions": {
     "boost": "1_66",
     "libcurl": "libcurl/7.61.1 OpenSSL/1.1.1g zlib/1.2.11 brotli/1.0.6 libidn2/2.2.0 libpsl/0.20.2 (+libidn2/2.2.0) libssh/0.9.4/openssl/zlib nghttp2/1.33.0",
     "libmicrohttpd": "0.9.70",
     "openssl": "1.1",
     "rapidjson": "1.1.0",
     "mongoc": "1.17.4",
     "bson": "1.17.4"
  }
}
}

[07:25:34] [OK] .............. The END-TO-END Check of the contextBroker is OK ..............

[07:25:34] [INFO] ######## Stopping the contextBroker service ########
Stopping...
contextBroker (pid  681403) is running...
Stopping contextBroker...                                  [  OK  ]
[07:25:35] [OK] .............. contextBroker service is stopped ..............

[07:25:35] [INFO] ######## Uninstall the contextBroker RPM ########
[07:25:37] [OK] .............. Uninstall completed! ..............

[07:25:37] [INFO] ######## Check if the directories are cleaned... ########
[07:25:37] [OK] .............. All clean ..............

[07:25:37] [INFO] ############ Check RPM Stage completed! ############
```

**Review by mapedraza — APPROVED** [review]
LGTM

**Pull request merged**